game/plugin/skills/herblore/src/potion.kt

```diff
@@ -0,0 +1,131 @@
+    DEFENCE(id = 133, unf_potion = UnfinishedPotion.RANARR, ingredient = NormalIngredient.WHITE_BERRIES.id, level = 30, experience = 75.0),
+    AGILITY(id = 3034, unf_potion = UnfinishedPotion.TOADFLAX, ingredient = NormalIngredient.TOADS_LEGS.id, level = 34, experience = 80.0),
+    PRAYER(id = 139, unf_potion = UnfinishedPotion.RANARR, ingredient = NormalIngredient.SNAPE_GRASS.id, level = 38, experience = 87.5),
+    SUPER_ATTACK(id = 145, unf_potion = UnfinishedPotion.IRIT, ingredient = NormalIngredient.EYE_NEWT.id, level = 45, experience = 100.0),
+    SUPER_ANTIPOISON(id = 181, unf_potion = UnfinishedPotion.IRIT, ingredient = CrushIngredient.UNICORN_HORN.crushed, level = 48, experience = 106.3),
+    FISHING(id = 151, unf_potion = UnfinishedPotion.AVANTOE, ingredient = NormalIngredient.SNAPE_GRASS.id, level = 50, experience = 112.5),
+    SUPER_ENERGY(id = 3018, unf_potion = UnfinishedPotion.AVANTOE, ingredient = NormalIngredient.MORT_MYRE_FUNGI.id, level = 52, experience = 117.5),
+    SUPER_STRENGTH(id = 157, unf_potion = UnfinishedPotion.KWUARM, ingredient = NormalIngredient.LIMPWURT_ROOT.id, level = 55, experience = 125.0),
+    WEAPON_POISON(id = 187, unf_potion = UnfinishedPotion.KWUARM, ingredient = CrushIngredient.DRAGON_SCALE.crushed, level = 60, experience = 137.5),
+    SUPER_RESTORE(id = 3026, unf_potion = UnfinishedPotion.SNAPDRAGON, ingredient = NormalIngredient.RED_SPIDERS_EGGS.id, level = 63, experience = 142.5),
+    SUPER_DEFENCE(id = 163, unf_potion = UnfinishedPotion.CADANTINE, ingredient = NormalIngredient.WHITE_BERRIES.id, level = 66, experience = 150.0),
+    ANTIFIRE(id = 2428, unf_potion = UnfinishedPotion.LANTADYME, ingredient = CrushIngredient.DRAGON_SCALE.crushed, level = 69, experience = 157.5),
+    RANGING(id = 169, unf_potion = UnfinishedPotion.DWARF_WEED, ingredient = NormalIngredient.WINE_ZAMORAK.id, level = 72, experience = 162.5),
+    MAGIC(id = 3042, unf_potion = UnfinishedPotion.LANTADYME, ingredient = NormalIngredient.POTATO_CACTUS.id, level = 76, experience = 172.5),
+    ZAMORAK_BREW(id = 189, unf_potion = UnfinishedPotion.TORSTOL, ingredient = NormalIngredient.JANGERBERRIES.id, level = 78, experience = 175.0);
+
+    val unf_potion = unf_potion.id
+    companion object {
+        private val FINISHED_POTION = FinishedPotion.values().associateBy  { Pair(it.unf_potion, it.ingredient) }
+        operator fun get(key: Pair<Int, Int>) = FINISHED_POTION[key]
+    }
+}
+
+class MakeFinishedPotionAction (val player: Player, val finished: FinishedPotion) : AsyncAction<Player>(1, false, player) {
+    override fun action(): ActionBlock = {
+        val level = mob.herblore.current
+
+        if (level < finished.level) {
+            player.sendMessage("You need an herblore level of ${finished.level} to make this potion.")
+            stop()
+        }
+
+        if (player.inventory.remove(finished.unf_potion, 1) > 0) {
```

Comment:
`player.inventory.remove(id, amount)` returns the number of items that was removed.  If you only want to remove one, you can write `player.inventory.remove(id)` intead, which returns a boolean indicating if one value was removed/.